mantra-game/plugins/gamepad/Gamepad.js

```diff
@@ -1,15 +1,73 @@
+import bitdoNesPro from './gamepads/8BitdoNES30Pro.js';
+import ps3 from './gamepads/Playstation3.js';
+import logitechDualAction from './gamepads/LogitechDualAction.js';
+const axesAssociation = {
+  'DPAD_HORIZONTAL' : ['DPAD_LEFT', 'DPAD_RIGHT'],
+  'DPAD_VERTICAL' : ['DPAD_UP', 'DPAD_DOWN'],
+  'LEFT_STICK_HORIZONTAL' : ['LEFT_STICK_LEFT', 'LEFT_STICK_RIGHT'],
+  'LEFT_STICK_VERTICAL' : ['LEFT_STICK_UP', 'LEFT_STICK_DOWN'],
+  'RIGHT_STICK_HORIZONTAL' : ['RIGHT_STICK_LEFT', 'RIGHT_STICK_RIGHT'],
+  'RIGHT_STICK_VERTICAL' : ['RIGHT_STICK_UP', 'RIGHT_STICK_DOWN']
+};
+
+const modules = import.meta.glob("./gamepads/*.js");
+
+
 export default class Gamepad {
 
   static id = 'gamepad';
+  static defaultControllerConfig = {
+    buttons : {
+        'BUTTON_A' : 0,
+        'BUTTON_B' : 2,
+        'BUTTON_X' : 3,
+        'BUTTON_Y' : 1,
+        'BUTTON_L1': 4,
+        'BUTTON_R1': 5,
+        'BUTTON_L2': 6,
+        'BUTTON_R2': 7,
+        // L3 & R3 aren't fixed 
+        // (sticks, touch pads, and other)
+        //'BUTTON_L3':8 : 'l2',
+        //'BUTTON_R3':9 : 'r2',
+        'BUTTON_SELECT':8,
+        'BUTTON_START':9,
+        'BUTTON_STICK_L' : 10,
+        'BUTTON_STICK_R' :11,
+    },
+    axes : {
+        'LEFT_STICK_HORIZONTAL' : 0,
+        'LEFT_STICK_VERTICAL' : 1,
+        'RIGHT_STICK_HORIZONTAL' : 2,
+        'RIGHT_STICK_VERTICAL' : 3,
+        'DPAD_HORIZONTAL' : 4,
+        'DPAD_VERTICAL' : 5,
+    }
+  };
 
   constructor() {
     this.id = Gamepad.id;
     this.gamepads = {};
+    this.configs = {
+      // prestuff the cache so no lookups happen to known controllers
+      'OEJpdGRvIE5FUzMwIFBybyAoVmVuZG9yOiAyZGM4IFByb2R1Y3Q6IDM4MjAp': bitdoNesPro,
```

Comment:
Dynamic loading and caching of external resources is not the responsibility of the `Gamepad`. There needs to zero code in this file related to dynamically loading or importing files.

Reply:
Corrected. That said, you'll find in order to have definitions that are directly loadable, you'll need a significantly different scheme.